### PR TITLE
Recipe app : Submission

### PR DESCRIPTION
**Pull request by Chrissiku**
## Project requirements

- You should follow the layout of the wireframes provided. You should personalize the rest of the design including colors, typographies, spacings, etc.
- Login page and registration page:
    - Should be built with Devise.
- Food list:
    - Should display a list of food added by the logged-in user as in the wireframe (for teams with only 2 people - display also quantity of a given food).
    - Should lead to a form that allows users to add new food.
- Recipes list:
    - Should display a list of recipes created by the logged-in user as in the wireframe.
    - Should lead to recipe details.
    - If the user is the owner of the recipe, should allow the user to delete it.
- Public recipe list:
    - Should display a list of all public recipes ordered by newest as in the wireframe.
    - Should lead to recipe details.
- Recipe details:
    - If the recipe is public or the user is the owner of the recipe, should display the recipe details as in the wireframe.
    - If the user is the owner of the recipe, should lead to the form that allows the user to add new food.
- General shopping list view (**only for a group that has less than 3 members**):
    - Should show the list of food that is missing for all recipes of the logged-in user (compare the list of food for all recipes with the general food list of that user).
    - Should count the total food items and total price of the missing food.
- Make sure there are no N+1 queries happening.
- Create a navigation menu that allows users to open all of the pages you created.

**If your team has 3 members these are requirements; otherwise, they are additional requirements**

- Inventories list:
    - Should display a list of inventories created by the logged-in user as in the wireframe.
    - Should lead to inventory details.
    - If the user is the owner of the inventory, should allow to delete it.
- Inventory details:
    - Should display the inventory details as in the wireframe.
    - Should lead to a form that allows users to add new food.
- Recipe details:
    - Should have a modal to choose an inventory to create a shopping list with, as in the wireframe.
- Inventory shopping list:
    - Should show the list of food that is missing by comparing the food in the recipe with the food in the inventory.
    - Should count the total food items and total price of the missing food.

**Technical setup**
- Set up the repository on GitHub and use Gitflow.
- Set up Devise for authentication.
- Set up RSpec and Capybara testing libraries.

## Show your support

Give a ⭐️ if you like this project!

### CODE REVIEW

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @Chrissiku, @Abdo9826 and @safar1212,

Good job so far!
✔️ Good documentation
✔️ Unit tests are passing
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
- [ ] You have done well including the unit tests. Kindly include integration tests using capybara

- [ ] _Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/recipes/show.html.erb [hunk 1]
```diff
@@ -0,0 +1,41 @@
+<div class="d-flex flex-column align-items-center justify-content-center p-5">
+    <h1><%= @recipe.name %></h1>
+    <div class="showContainer">
+    <div>
+        <p>Preparation Time : <%= @recipe.preperation_time %></p>
+        <p>Coocking time : <%= @recipe.cooking_time %></p>
+        <%= link_to "Generate shopping list", user_recipes_path %>
+    </div>
```
[the remark this answers is not in the document]
Comment (safar1212) [reply app/views/recipes/show.html.erb:8]: Dear Reviewer,
@TedLivist, Well the general shopping list is for a group of 2 people. We are a group of 3 people. And that is not a
 requirement for our group. Please take a look at the below screenshot.

[the remark this answers is not in the document]
Comment (TedLivist) [reply app/views/recipes/show.html.erb:8]: Hi @safar1212 alright. I'll update it to OPTIONAL then.

[the remark this answers is not in the document]
Comment (safar1212) [reply app/views/recipes/show.html.erb:8]: Thanks alot @TedLivist

**Review by descholar-ceo — APPROVED** [review]
Hi @Chrissiku && @safar1212 🤝 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
👍🏽 Implemented all requested changes ✔️ 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**